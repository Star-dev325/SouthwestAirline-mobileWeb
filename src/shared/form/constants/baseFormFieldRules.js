import validator from 'src/shared/form/formValidators/validator';
import i18n from '@swa-ui/locale';

const isRequired = true;

export const baseFieldRules = {
  addressLine1: [
    {
      isRequired
    },
    {
      msg: i18n('SHARED__ERROR_MESSAGES__ADDRESS_LINE_1'),
      validator: validator.isAddress
    },
    {
      msg: i18n('SHARED__ERROR_MESSAGES__ADDRESS_LINE_1_LENGTH'),
      validator: validator.isLengthLessOrEqual(40)
    }
  ],
  addressLine2: [
    {
      msg: i18n('SHARED__ERROR_MESSAGES__ADDRESS_LINE_2'),
      validator: validator.isAddress
    },
    {
      msg: i18n('SHARED__ERROR_MESSAGES__ADDRESS_LINE_2_LENGTH'),
      validator: validator.isLengthLessOrEqual(40)
    }
  ],
  city: [
    {
      isRequired
    },
    {
      msg: i18n('SHARED__ERROR_MESSAGES__CITY'),
      validator: validator.isCity
    }
  ]
};

export const basePostal = {
  stateProvince: [
    {
      msg: i18n('SHARED__ERROR_MESSAGES__STATE_PROVINCE_REGION'),
      validator: validator.isStateProvinceRegion
    },
    {
      msg: i18n('SHARED__ERROR_MESSAGES__STATE_PROVINCE_REGION_LENGTH'),
      validator: validator.isLengthLessThan(50)
    }
  ],
  isUSPostal: [
    {
      msg: i18n('SHARED__ERROR_MESSAGES__ZIP_FOR_US'),
      validator: (value) => validator.isNumeric(value) && validator.isLengthEql(5)(value)
    }
  ],
  postalCode: [
    {
      msg: i18n('SHARED__ERROR_MESSAGES__POSTAL_CODE'),
      validator: validator.isPostalCode
    },
    {
      msg: i18n('SHARED__ERROR_MESSAGES__POSTAL_CODE_LENGTH'),
      validator: validator.isLengthLessOrEqual(10)
    }
  ]
};
