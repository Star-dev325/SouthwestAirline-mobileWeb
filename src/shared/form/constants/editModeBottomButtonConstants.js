export default {
  PRIMARY: 'PRIMARY',
  UPDATE: 'UPDATE',
  DELETE: 'DELETE',
  buttonText: {
    PRIMARY: 'Primary',
    UPDATE: 'Update',
    DELETE: 'Delete'
  }
};
