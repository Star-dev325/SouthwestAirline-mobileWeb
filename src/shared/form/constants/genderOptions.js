export default {
  Male: 'Male',
  Female: 'Female'
};
