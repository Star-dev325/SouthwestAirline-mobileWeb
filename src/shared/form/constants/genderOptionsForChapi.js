export default {
  M: 'Male',
  F: 'Female'
};
