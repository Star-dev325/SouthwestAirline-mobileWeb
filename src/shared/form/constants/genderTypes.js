export default {
  FEMALE: 'F',
  MALE: 'M',
  MALE_FULL: 'MALE',
  FEMALE_FULL: 'FEMALE',
  UNAVAILABLE: 'UNAVAILABLE'
};
