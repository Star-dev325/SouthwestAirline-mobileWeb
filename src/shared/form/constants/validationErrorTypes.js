export const REQUIRED_ERROR = 'REQUIRED_ERROR';
export const FIELD_ERROR_MESSAGE = 'FIELD_ERROR_MESSAGE';
export const ERROR_HEADER = 'ERROR_HEADER';
export const SIMPLE_ERROR_POPUP = 'SIMPLE_ERROR_POPUP';
export const API_ERROR_POPUP = 'API_ERROR_POPUP';

export const DEFAULT_ERROR_TYPE = FIELD_ERROR_MESSAGE;
