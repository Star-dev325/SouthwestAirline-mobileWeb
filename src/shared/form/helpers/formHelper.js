// @flow
const repeatStr = (str, times = 1) => {
  let result = str;

  while (times > 1) {
    result += str;
    times--;
  }

  return result;
};

export const getMaskProps = function (mask: string | { rule: string, repeat?: number, maskChar?: string }) {
  const defaultMask = {
    repeat: 1,
    maskChar: null
  };

  const maskConfigs = Object.assign({}, defaultMask, typeof mask === 'string' ? { rule: mask } : mask);

  const finalMaskStr = repeatStr(maskConfigs.rule, maskConfigs.repeat);

  return {
    mask: finalMaskStr,
    maskChar: maskConfigs.maskChar,
    maxLength: finalMaskStr.length // mask can't work properly in some android device so use maxLength to limit the user input, see bug#2026
  };
};
