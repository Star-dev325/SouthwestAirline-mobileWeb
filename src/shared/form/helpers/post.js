import _ from 'lodash';
import BrowserObject from 'src/shared/helpers/browserObject';

const { document } = BrowserObject;

const post = (formProps, params) => {
  const createForm = () => {
    const form = document.createElement('form');
    const { target } = formProps;

    form.setAttribute('method', formProps.method || 'POST');
    form.setAttribute('action', formProps.path);

    if (target) {
      form.setAttribute('target', target);
    }

    return form;
  };

  const createSubmitButton = () => {
    const submitButton = document.createElement('input');

    submitButton.setAttribute('type', 'submit');
    submitButton.setAttribute('style', 'display: none;');
    submitButton.setAttribute('value', 'submitButton');

    return submitButton;
  };

  const createInput = (key, value) => {
    const input = document.createElement('input');

    input.setAttribute('type', 'hidden');
    input.setAttribute('name', key);
    input.setAttribute('value', value);

    return input;
  };

  if (formProps && formProps.path) {
    const form = createForm();

    _.each(params, (value, key) => {
      form.appendChild(createInput(key, params[key]));
    });

    form.appendChild(createSubmitButton());
    document.body.appendChild(form);
    form.submit();
    form.parentNode.removeChild(form);
  }
};

export default post;
