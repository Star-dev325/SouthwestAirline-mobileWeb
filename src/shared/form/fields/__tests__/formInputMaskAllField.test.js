import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { createMockedFormStore, createMockedForm } from 'test/unit/helpers/formTestUtils';
import FormInputMaskAllField from 'src/shared/form/fields/formInputMaskAllField';

import { FIELD_ERROR_MESSAGE } from 'src/shared/form/constants/validationErrorTypes';

describe('FormInputMaskAllField', () => {
  let onChangeStub;
  let onSubmitStub;
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('when invalid', () => {
    it('should show error message', () => {
      const { container } = createComponentWithInvalidState();

      expect(container).toMatchSnapshot();
    });

    it('should clear the value when focus', () => {
      const { container } = createComponentWithInvalidState();

      fireEvent.focus(container.querySelector('input'));
      fireEvent.submit(container.querySelector('form'));

      expect(onSubmitStub).toHaveBeenCalledWith({ name: '' });
    });

    it('should display red error exclamation icon when input field value is invalid', () => {
      const { container } = createComponentWithInvalidState({ showWarningIcon: true });

      expect(container.querySelector('.icon_exclamation-circle')).not.toBeNull();
    });

    it('should not display red error exclamation icon when input field value is invalid but noErrorIcon is true', () => {
      const { container } = createComponentWithInvalidState({ noErrorIcon: true });

      expect(container.querySelector('.icon_exclamation-circle')).toBeNull();
    });
  });

  describe('icon', () => {
    it('should display blue exclamation icon when input field is empty', () => {
      const { container } = createComponent({ showWarningIcon: true }, undefined, '');

      expect(container.querySelector('.icon_exclamation-circle')).not.toBeNull();
    });

    it('should not display blue warning exclamation icon when input field is not empty', () => {
      const { container } = createComponent({ showWarningIcon: true, value: '12345' });

      expect(container.querySelector('.icon_exclamation-circle')).toBeNull();
    });

    it('should not display blue warning icon when showWarningIcon is not passed', () => {
      const { container } = createComponent();

      expect(container.querySelector('.icon_exclamation-circle')).toBeNull();
    });

    it('should not have warning icon when input field is valid and showWarningIcon is false', () => {
      const { container } = createComponent({ showWarningIcon: false });

      expect(container.querySelector('.icon_exclamation-circle')).toBeNull();
    });
  });

  describe('masking', () => {
    it('should display masked value when input field is not empty', () => {
      const { container } = createComponent({ value: '1234' });

      expect(container.querySelector('.mask-field')).not.toBeNull();
    });

    it('should display unmasked value when field has focus', () => {
      const { container } = createComponent({ value: '123' });
      const input = container.querySelector('input');

      fireEvent.focus(input);

      expect(input.value).toEqual('123');
    });
  });

  describe('extra props', () => {
    it('should have extra props when extra props are passed into FormInputFieldMaskAll', () => {
      const { container } = createComponent({ type: 'tel' });

      expect(container.querySelector('input').type).toEqual('tel');
    });
  });

  function createComponentWithInvalidState(props) {
    const validatorStub = jest
      .fn()
      .mockReturnValueOnce({ name: { type: FIELD_ERROR_MESSAGE, msg: 'something wrong' } })
      .mockReturnValueOnce({});
    const { container, unmount } = createComponent(props, {
      formValidator: () => validatorStub,
      defaultValues: () => ({
        name: ''
      })
    });

    fireEvent.submit(container.querySelector('form'));

    return { container, unmount };
  }

  function createComponent(props, formOptions, initialValue = 'init value') {
    onSubmitStub = jest.fn();
    onChangeStub = jest.fn();
    store = createMockedFormStore();
    const MockedForm = createMockedForm(store, formOptions);

    return render(
      <MockedForm initialFormData={{ name: initialValue }} onSubmit={onSubmitStub}>
        <FormInputMaskAllField name="name" value="" maskChar="*" onChange={onChangeStub} {...props} />
      </MockedForm>
    );
  }
});
