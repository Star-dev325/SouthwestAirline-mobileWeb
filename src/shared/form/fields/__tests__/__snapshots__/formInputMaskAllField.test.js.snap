// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputMaskAllField when invalid should show error message 1`] = `
<div>
  <form>
    <div
      class="field error"
    >
      <div
        class="mask-field input"
      >
        <input
          class=""
          icon="exclamation-circle"
          name="name"
          value=""
        />
        <i
          class="icon icon_exclamation-circle"
        />
      </div>
      <span
        class="field--error-msg"
      >
        something wrong
      </span>
    </div>
  </form>
</div>
`;
